src/Components/MSBuild.fs

```diff
@@ -24,7 +24,7 @@ module MSBuild =
             | Ok msbuild -> Promise.lift msbuild
             | Error msg -> Promise.reject (exn msg))
 
-    let invokeMSBuildWithCancel project target (cancellationToken: CancellationToken) =
+    let invokeMSBuildWithCancel project target (cancellationToken: CancellationToken) args =
```

Comment:
We're doing the classic calling `msbuild t:Build` here instead of `dotnet build` which can cause issues with needing to restore. This is currently in a hacky state so we can discuss options.

Reply:
on thing to consider is that `/t:Build` calls the build target - and not all projects will have that target. Most _managed_ projects will, but what `dotnet build` and `msbuild` do by default is call the _default target_ so the most correct thing to do might be to allow specifying an optional target?

you can also do `dotnet build` with `--no-restore` if you want to skip the implicit restore.